src/main/java/com/atlauncher/gui/LauncherFrame.java

```diff
@@ -160,6 +165,37 @@ public void componentMoved(ComponentEvent evt) {
         });
     }
 
+    /**
+     * There are still tasks running,
+     * ask if the user realllly wants to close.
+     */
+    private void askToClose() {
+        int response = DialogManager.yesNoDialog()
+            .setContent(GetText.tr("There are tasks still running, are you sure you want to exit?"))
+            .show();
+
+        if (response == 0) {
+            close();
+        }
+    }
+
+    /**
+     * Close the application.
+     */
+    private void close() {
+        WorkerManager.stopAll();
```

Comment:
For later consideration, if user does this while a pack is installing or say an instance updating, need to make sure there is something that is done to those instances as they are being stopped in a bad state, so marking them as invalid or simply deleting the unfinished instance folder. Maybe just a way for a task to be able to define a cleanup

Reply:
Currently, `LauncherFrame.askToClose` exists to somewhat prevent this if something is running.

I can add a `onStopped` to `Worker` instances.